esque/cli/commands.py

```diff
@@ -112,58 +113,56 @@ def edit_topic(state: State, topic_name: str):
-        for topic in topics_to_be_changed
-    }
-
-    if len(topic_config_diffs) > 0:
-        click.echo(pretty_topic_diffs(topic_config_diffs))
-        if ensure_approval("Are you sure to change configs?", no_verify=state.no_verify):
-            topic_controller.alter_configs(topics_to_be_changed)
-            to_change = [topic.name for topic in topics_to_be_changed]
-            click.echo(
-                click.style(
-                    pretty({"Successfully changed topics": to_change}), fg="green"
-                )
-            )
-    else:
-        click.echo("No topics to edit.")
-
-    new_topics = [topic for topic in new_topic_configs if topic not in editable_topics]
-    if len(new_topics) > 0:
-        click.echo(get_output_new_topics(new_topics))
-        if ensure_approval("Are you sure to create the new topics?", no_verify=state.no_verify):
-            topic_controller.create_topics(new_topics)
-            to_create = [topic.name for topic in new_topics]
-            click.echo(
-                click.style(
-                    pretty({"Successfully created topics": to_create}), fg="green"
-                )
-            )
-    else:
-        click.echo("No new topics to create.")
+    cluster_topics = topic_controller.list_topics(search_string="|".join(yaml_topic_names))
+    cluster_topic_names = [t.name for t in cluster_topics]
+
+    # Calculate changes
+    to_create = [yaml_topic for yaml_topic in yaml_topics if yaml_topic.name not in cluster_topic_names]
+    to_edit = [yaml_topic for yaml_topic in yaml_topics if
+               yaml_topic not in to_create and topic_controller.diff_with_cluster(yaml_topic) != {}]
+    to_edit_diffs = {t.name: topic_controller.diff_with_cluster(t) for t in to_edit}
+    to_ignore = [yaml_topic for yaml_topic in yaml_topics if yaml_topic not in to_create and yaml_topic not in to_edit]
+
+    # Sanity check - the 3 groups of topics should be complete and have no overlap
+    assert set(to_create).isdisjoint(set(to_edit)) \
+           and set(to_create).isdisjoint(set(to_ignore)) \
+           and set(to_edit).isdisjoint(set(to_ignore)) \
+           and len(to_create) + len(to_edit) + len(to_ignore) == len(yaml_topics)
+
+    # Print diffs so the user can check
+    click.echo(pretty_unchanged_topic_configs(to_ignore))
+    click.echo(pretty_new_topic_configs(to_create))
+    click.echo(pretty_topic_diffs(to_edit_diffs))
+
+    # Check for actionable changes
+    if len(to_edit) + len(to_create) == 0:
+        click.echo("No changes detected, aborting")
+        return
+
+    # Get approval
+    if not ensure_approval("Apply changes?", no_verify=state.no_verify):
+        click.echo("Cancelling changes")
+        return
+
+    # apply changes
+    topic_controller.create_topics(to_create)
+    topic_controller.alter_configs(to_edit)
+
+    # output confirmation
+    changes = {"ignored": len(to_ignore), "created": len(to_create), "changed": len(to_edit)}
```

Comment:
"Unchanged" is probably better than "ignored", otherwise I'd ask myself why they got ignored.

Reply:
agreed, fixed